lib/core/wireprotocol/command.js

```diff
@@ -45,14 +45,20 @@ function _command(server, ns, cmd, options, callback) {
   const shouldUseOpMsg = supportsOpMsg(server);
   const session = options.session;
 
-  let clusterTime = server.clusterTime;
+  const serverClusterTime = server.clusterTime;
+  let clusterTime = serverClusterTime;
   let finalCmd = Object.assign({}, cmd);
   if (hasSessionSupport(server) && session) {
+    const sessionClusterTime = session.clusterTime;
     if (
-      session.clusterTime &&
-      session.clusterTime.clusterTime.greaterThan(clusterTime.clusterTime)
+      serverClusterTime &&
+      serverClusterTime.clusterTime &&
+      sessionClusterTime &&
+      sessionClusterTime.clusterTime
     ) {
-      clusterTime = session.clusterTime;
+      if (sessionClusterTime.clusterTime.greaterThan(serverClusterTime.clusterTime)) {
```

Comment:
This may just be my own bias against nested if's, but why not just append that check to the end of the chain of the preceding 4 `&&` conditions?

Reply:
I agree, I'll fix